### ISSUE TITLE
Submission: <17: hotdog-predictor>

### ISSUE THREAD

**Issue by jentsang** [UBC-MDS/data-analysis-review-2025#3]
**Submitting authors:**  @hpalafoxp @samrawitt12 @zakiaslam22 @jentsang

**Repository:** https://github.com/jentsang/hotdog-predictor 
**Report link:** https://github.com/jentsang/hotdog-predictor/blob/main/reports/dog_or_not_report.pdf 

**Abstract/executive summary:**
In this project, we used decision trees, logistic regression, and a Naive Bayes classifier to identify whether or not a food vendor sells hot dogs via their name. We trained each model individually using a cross-validation setup, and we compared the scores of the accuracy in order to determine a model to train and to compare to the test data. The model we chose, finally, was the Naive Bayes, as it provided a slightly better score and less underfit and overfit than the other models present. Finally, we validated it with our test data and came to the conclusion that even though it is good enough for classifying most of the cases, it still struggles to discern from the minority class, which, in our case, is our target.

**Editor:** @jentsang
**Reviewer:** < JOEL PETERSON > < RAGHAV GUPTA > < @risha-daz > < GODSGIFT BRAIMAH >

- [X] I agree to abide by [MDS's Code of Conduct](https://ubc-mds.github.io/resources_pages/code_of_conduct/) during the review process and in maintaining my package should it be accepted.

**Comment by Brymahh** [comment]
## Data analysis review checklist

### Reviewer: @Brymahh
### Conflict of interest

- [x] As the reviewer I confirm that I have no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [MDS code of conduct](https://ubc-mds.github.io/resources_pages/code_of_conduct/).

### General checks

- [x] **Repository:** Is the source code for this data analysis available? Is the repository well organized and easy to navigate?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?

### Documentation

- [x] **Installation instructions:** Is there a clearly stated list of dependencies? 
- [x] **Example usage:** Do the authors include examples of how to use the software to reproduce the data analysis?
- [x] **Functionality documentation:** Is the core functionality of the data analysis software documented to a satisfactory level?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Code quality
- [x] **Readability:** Are scripts, functions, objects, etc., well named? Is it relatively easy to understand the code?
- [x] **Style guidelides:** Does the code adhere to well known language style guides?
- [x] **Modularity:** Is the code suitably abstracted into scripts and functions?
- [ ] **Tests:** Are there automated tests or manual steps described so that the function of the software can be verified? Are they of sufficient quality to ensure software robsutness?

### Reproducibility
 
- [x] **Data:** Is the raw data archived somewhere? Is it accessible?
- [x] **Computational methods:** Is all the source code required for the data analysis available?
- [x] **Conditions:** Is there a record of the necessary conditions (software dependencies) needed to reproduce the analysis? Does there exist an easy way to obtain the computational environment needed to reproduce the analysis?
- [x] **Automation:** Can someone other than the authors easily reproduce the entire data analysis?

### Analysis report

- [x] **Authors:** Does the report include a list of authors with their affiliations?
- [x] **What is the question:** Do the authors clearly state the research question being asked?
- [ ] **Importance:** Do the authors clearly state the importance for this research question?
- [ ] **Background**: Do the authors provide sufficient background information so that readers can understand the report?
- [x] **Methods:** Do the authors clearly describe and justify the methodology used in the data analysis? Do the authors communicate any assumptions or limitations of their methodologies?
- [x] **Results:** Do the authors clearly communicate their findings through writing, tables and figures?
- [x] **Conclusions:** Are the conclusions presented by the authors correct? 
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?
- [x] **Writing quality:** Is the writing of good quality, concise, engaging? 

### Estimated hours spent reviewing: 46 minutes

### Review Comments: 
1.
-------
Hello, <br>
Your ReadME  is absolutely detailed, but I think listing out the project folder structure might be too lengthy. I had to scroll all the way down to find the actual guidelines for running your scripts. I think you can use Sky’s structure to add collapsible sections for readability sake. I am attaching a link to the ReadMe here (You can take note of the collapsible triangles): https://github.com/skysheng7/ia4

2. <br>
-----
Hi again, I was struggling to find the answer to this question in your ReadMe `Importance: Do the authors clearly state the importance for this research question?` and then I checked the analysis file, your team mentioned: `In this project, we study whether we can predict if a food vendor is a hot dog vendor or not using the vendor’s business name.`  Since you didn't use only the Business Name, I think you can better rephrase this.

3.
--------
Hello,
For this review question: `Do the authors provide sufficient background information so that readers can understand the report?`, I was a bit unclear on why it can `be useful to automatically identify what kind of food a vendor specializes in based only on select information.(copied from your analysis report)` 
Does this save time? Or help staffs or people better order hotdogs? Or is there an added advantage to the food vendors. I think you could expatiate on the value. 

<br>
Overall well done to your group. Your functions/scripts are absolutely detailed and follow real industry standards, I fancied the script introduction at the top. Best of luck!🤗


#### Attribution

This was derived from the [JOSE review checklist](https://openjournals.readthedocs.io/en/jose/review_checklist.html) and the ROpenSci review checklist.

**Comment by risha-daz** [comment]
## Data analysis review checklist

### Reviewer: @risha-daz

### Conflict of interest

- [x] As the reviewer I confirm that I have no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [MDS code of conduct](https://ubc-mds.github.io/resources_pages/code_of_conduct/).

### General checks

- [x] **Repository:** Is the source code for this data analysis available? Is the repository well organized and easy to navigate?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?

### Documentation

- [x] **Installation instructions:** Is there a clearly stated list of dependencies? 
- [x] **Example usage:** Do the authors include examples of how to use the software to reproduce the data analysis?
- [x] **Functionality documentation:** Is the core functionality of the data analysis software documented to a satisfactory level?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Code quality
- [x] **Readability:** Are scripts, functions, objects, etc., well named? Is it relatively easy to understand the code?
- [x] **Style guidelides:** Does the code adhere to well known language style guides?
- [x] **Modularity:** Is the code suitably abstracted into scripts and functions? -> Scripts are very good!
- [ ] **Tests:** Are there automated tests or manual steps described so that the function of the software can be verified? Are they of sufficient quality to ensure software robsutness?

### Reproducibility
 
- [x] **Data:** Is the raw data archived somewhere? Is it accessible?
- [x] **Computational methods:** Is all the source code required for the data analysis available?
- [ ] **Conditions:** Is there a record of the necessary conditions (software dependencies) needed to reproduce the analysis? Does there exist an easy way to obtain the computational environment needed to reproduce the analysis? (Doesn't include instructions to generate the literal document)
- [x] **Automation:** Can someone other than the authors easily reproduce the entire data analysis?

### Analysis report

- [x] **Authors:** Does the report include a list of authors with their affiliations?
- [x] **What is the question:** Do the authors clearly state the research question being asked?
- [x] **Importance:** Do the authors clearly state the importance for this research question?
- [x] **Background**: Do the authors provide sufficient background information so that readers can understand the report?
- [x] **Methods:** Do the authors clearly describe and justify the methodology used in the data analysis? Do the authors communicate any assumptions or limitations of their methodologies?
- [x] **Results:** Do the authors clearly communicate their findings through writing, tables and figures?
- [x] **Conclusions:** Are the conclusions presented by the authors correct? 
- [ ] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)? (W3Schools reference is not clear what it is pointing at)
- [x] **Writing quality:** Is the writing of good quality, concise, engaging? 

### Estimated hours spent reviewing: 

### Review Comments: 

1. Very creative topic! And the corresponding analysis is very sound.
2. The graphs and flow of the report is very good. Takes the time to make the reader familiar with the data. I like the notebook even more because it spends more time in this and would love to see some of the exploration from there in the qmd, but i understand the qmd wants to be concise. Could take a second look at this if you feel the same!
3. Would love to see more instructions on generating the literal document! The documentation expects the user to use the notebook to dynamically read the analysis instead of a literal document. It doesn't have instructions to render the literal documents, and the user is supposed to use the existing documents to read the reports. Also, not necessary but would love to see the html rendered on github! Since the html file has supplementary metadata only downloading and viewing it doesnt work. Also `Docker` wasn't listed as a dependency.
4. Some of the column names and text inside the literal document are using the variable name directlyeg `DummyClassifier` `max_features` `GEO_LOCALAREA` `CSV` `DataFrame`. Makes it a bit slower read! 
5. A small nitpick but the introduction has a lot of details about the dataset, my opinion would be to just introduce the data superficially here and give more details in the methods: data section
6. Another small nitpick: Could divide the analysis and EDA section into paragraphs for faster read! 
7. could change the table titles to be readable! It has a few stray nan values too. Also I feel table 6 and 8 doesn't need to show all that information. As a reader I could not understand the probabilities column since it had so many numbers. I would suggest to round them or remove them, or divide that column into two and color code the cells (like you did for correlation)
8. Footnote: agree with @Brymahh The documentation is a bit long and contains some information that doesn't feel useful. It also has the instructions to use the ipynb at the beginning, which implies that it is the primary way of interacting with the project. Also the notebook and the report have different content.
9. The W3schools reference does not make sense in the flow of the document. Please check if it could be clearer why it is being referenced and what exactly is it referenced for.

#### Attribution

This was derived from the [JOSE review checklist](https://openjournals.readthedocs.io/en/jose/review_checklist.html) and the ROpenSci review checklist.

**Comment by j031nich0145** [comment]
## Data analysis review checklist

### Reviewer: Github Username: <@ j031nich0145>  IRL:  Joel Peterson

### Conflict of interest

- [ ✅ ] As the reviewer I confirm that I have no conflicts of interest for me to review this work.

### Code of Conduct

- [✅  ] I confirm that I read and will adhere to the [MDS code of conduct](https://ubc-mds.github.io/resources_pages/code_of_conduct/).

### General checks

- [ ✅ ] **Repository:** Is the source code for this data analysis available? Is the repository well organized and easy to navigate?
- [✅  ] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?

### Documentation

- [✅  ] **Installation instructions:** Is there a clearly stated list of dependencies? 
- [ ✅ ] **Example usage:** Do the authors include examples of how to use the software to reproduce the data analysis?
- [✅  ] **Functionality documentation:** Is the core functionality of the data analysis software documented to a satisfactory level?
- [✅  ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Code quality
- [✅  ] **Readability:** Are scripts, functions, objects, etc., well named? Is it relatively easy to understand the code?
- [ ✅ ] **Style guidelides:** Does the code adhere to well known language style guides?
- [✅  ] **Modularity:** Is the code suitably abstracted into scripts and functions?
- [ ⏱️] **Tests:** Are there automated tests or manual steps described so that the function of the software can be verified? Are they of sufficient quality to ensure software robsutness? 
- ### ❗Tests were not part of Milestone 3 so I can anticipate some fantastic tests are on the way.

### Reproducibility
 
- [ ✅] **Data:** Is the raw data archived somewhere? Is it accessible? 
- ### ❗Everything is so organized and in the right place in the repo => just a small (tiny) note that the link for the data 'here' in the report pdf under the data header does not produce a hyperlink as maybe expected :)
- [ ✅] **Computational methods:** Is all the source code required for the data analysis available?
- [✅ ] **Conditions:** Is there a record of the necessary conditions (software dependencies) needed to reproduce the analysis? Does there exist an easy way to obtain the computational environment needed to reproduce the analysis?
- [✅ ] **Automation:** Can someone other than the authors easily reproduce the entire data analysis?
### ❗Really thorough documentation, organization, guidelines and instructions to reproduce the workflow!

### Analysis report

- [ ✅ ] **Authors:** Does the report include a list of authors with their affiliations?
- [ ✅ ] **What is the question:** Do the authors clearly state the research question being asked?
- [ ✅ ] **Importance:** Do the authors clearly state the importance for this research question?
- [  ✅ ] **Background**: Do the authors provide sufficient background information so that readers can understand the report?
-   [ ✅] **Methods:** Do the authors clearly describe and justify the methodology used in the data analysis? Do the authors communicate any assumptions or limitations of their methodologies?
- [  ✅ ] **Results:** Do the authors clearly communicate their findings through writing, tables and figures?
### ❗Just another tiny note ! Some of the figure renderings in the pdf are a bit off with titles and sizing and some information being cropped etc => but of course these things happen and can be quite unpredictable! Just a note :)
- [  ✅ ] **Conclusions:** Are the conclusions presented by the authors correct? 
- [  ✅ ]  **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?
- [ ✅] **Writing quality:** Is the writing of good quality, concise, engaging? 

### Estimated hours spent reviewing: ~1hr

### ❗Review Comments: 

Really great job! This was a extensive and thorough report. The repository is elegantly organized with all files and folders arranged intuitively with clear license, code of conduct/contribution, and reproducibility instructions. The report is very thorough but concise with an impressive amount of figures and analysis. The only finicky improvements to make would likely be a bit of trial and error with the pdf renders. Overall a very well put together analysis.


#### Attribution

This was derived from the [JOSE review checklist](https://openjournals.readthedocs.io/en/jose/review_checklist.html) and the ROpenSci review checklist.

**Comment by raghav9048** [comment]
## Data analysis review checklist

### Reviewer: <raghav9048> 

### Conflict of interest

- [x ] As the reviewer I confirm that I have no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [MDS code of conduct](https://ubc-mds.github.io/resources_pages/code_of_conduct/).

### General checks

- [x] **Repository:** Is the source code for this data analysis available? Is the repository well organized and easy to navigate?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?

### Documentation

- [x] **Installation instructions:** Is there a clearly stated list of dependencies? 
- [x] **Example usage:** Do the authors include examples of how to use the software to reproduce the data analysis?
- [x] **Functionality documentation:** Is the core functionality of the data analysis software documented to a satisfactory level?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Code quality
- [x] **Readability:** Are scripts, functions, objects, etc., well named? Is it relatively easy to understand the code?
- [x] **Style guidelides:** Does the code adhere to well known language style guides?
- [x] **Modularity:** Is the code suitably abstracted into scripts and functions?
- [ ] **Tests:** Are there automated tests or manual steps described so that the function of the software can be verified? Are they of sufficient quality to ensure software robsutness? raghav9048 : I don't see automated tests but I think it was not explicitly mentioned to have them till milestone 3.

### Reproducibility
 
- [x] **Data:** Is the raw data archived somewhere? Is it accessible?
- [x] **Computational methods:** Is all the source code required for the data analysis available?
- [x] **Conditions:** Is there a record of the necessary conditions (software dependencies) needed to reproduce the analysis? Does there exist an easy way to obtain the computational environment needed to reproduce the analysis?
- [x] **Automation:** Can someone other than the authors easily reproduce the entire data analysis?

### Analysis report

- [x] **Authors:** Does the report include a list of authors with their affiliations?
- [x] **What is the question:** Do the authors clearly state the research question being asked?
- [x] **Importance:** Do the authors clearly state the importance for this research question?
- [x] **Background**: Do the authors provide sufficient background information so that readers can understand the report?
- [x] **Methods:** Do the authors clearly describe and justify the methodology used in the data analysis? Do the authors communicate any assumptions or limitations of their methodologies?
- [x] **Results:** Do the authors clearly communicate their findings through writing, tables and figures?
- [x] **Conclusions:** Are the conclusions presented by the authors correct? 
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?
- [x] **Writing quality:** Is the writing of good quality, concise, engaging? 

### Estimated hours spent reviewing: 1.5

The project is very well organized, reproducible & clearly documented from data validation through model selection & evaluation. The report is easy to follow & shows good reasoning behind the modelling choices. The repository structure is clean, with raw/processed data, scripts, notebooks and reports separated appropriately.

### Review Comments: 
1. The hyperparameter search results table has overlapping column labels in the PDF output, making it difficult to read. Renaming columns to shorter labels or displaying only the key hyperparameter columns (max_features and alpha) would make the table more readable.
2. There are multiple minor spelling mistakes in the report (eg., "preformed" instead of "performed", "anomolies" instead of "anomalies"). 
3. The caption describing the hyperparameter tuning table refers to Logistic Regression, but the tuning was applied to Naive Bayes. Updating the caption would prevent confusion for other readers.
4. In the EDA section, some of the visualizations suffer from label overlap and overplotting.
5. There are no automated tests in the repository. However, I believe that up to Milestone 3, automated testing was not an explicit requirement, so it makes sense that they are not included at this stage. I am simply noting their absence, not indicating that this is a mistake.
6. Overall, I think this is a strong & reproducible project with a clear workflow. Implementing the improvements above would make the report even stronger & more polished.

#### Attribution

This was derived from the [JOSE review checklist](https://openjournals.readthedocs.io/en/jose/review_checklist.html) and the ROpenSci review checklist.
